ln-gateway/src/lib.rs

```diff
@@ -365,3 +368,11 @@ pub fn serde_hex_deserialize<'d, T: bitcoin::consensus::Decodable, D: Deserializ
             .map_err(|e| serde::de::Error::custom(format!("{:?}", e)))
     }
 }
+
+impl IntoResponse for LnGatewayError {
+    fn into_response(self) -> Response {
+        let mut err = Cow::<'static, str>::Owned(format!("{:?}", self)).into_response();
+        *err.status_mut() = StatusCode::INTERNAL_SERVER_ERROR;
```

Comment:
Eventually we'll want to be smarter about the status code we return. 

For example, if it can't find a contract we should return a 404 and not a 500. But I don't think we do this yet. So this is good for now.